apps/explorer/src/components/Activity/CheckpointsActivityTable.tsx

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { ArrowRight12 } from '@mysten/icons';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo, useState } from 'react';
+
+import { genTableDataFromCheckpointsData } from './utils';
+
+import { useGetCheckpoints } from '~/hooks/useGetCheckpoints';
+import { Link } from '~/ui/Link';
+import { Pagination } from '~/ui/Pagination';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { TableCard } from '~/ui/TableCard';
+import { Text } from '~/ui/Text';
+import { numberSuffix } from '~/utils/numberUtil';
+
+const DEFAULT_CHECKPOINTS_LIMIT = 20;
+
+interface Props {
+    disablePagination?: boolean;
+    refetchInterval?: number;
+    initialLimit?: number;
+    initialCursor?: string;
+    maxCursor?: string;
+}
+
+export function CheckpointsActivityTable({
+    disablePagination,
+    initialLimit = DEFAULT_CHECKPOINTS_LIMIT,
+    initialCursor,
+    maxCursor,
+}: Props) {
+    const [currentPage, setCurrentPage] = useState(0);
+    const [limit, setLimit] = useState(initialLimit);
+    const rpc = useRpcClient();
+    const {
+        data,
+        isLoading,
+        isFetching,
+        isFetchingNextPage,
+        fetchNextPage,
+        hasNextPage,
+    } = useGetCheckpoints(initialCursor, limit);
+
+    const countQuery = useQuery(['checkpoints', 'count'], () =>
+        rpc.getLatestCheckpointSequenceNumber()
+    );
+
+    const count = useMemo(() => {
+        if (maxCursor && initialCursor)
```

Comment:
I'm confused about what we're doing with the max cursor here, where does this come into play? Seems like this maybe should just disable the count on this table in this case?

Reply:
we can display the count for number of checkpoints in an epoch using the max/initial comparison, otherwise we just use the count query data